### PR TITLE
Add SecondaryFiles expression tests

### PR DESCRIPTION
**Pull request by illusional**
Relates to: https://github.com/common-workflow-language/cwltool/pull/1233

Tests pass with my CWLTool modification in linked PR.

```bash
cwltest --test tests/secondaryfiles/test-index.yml --tool cwltool

# Test [1/2] Confirm CommandInputParameter expression can receive a File object
# Test [2/2] Confirm CommandOutputParameter expression can receive a File object
# All tests passed
```

**Related discussion:**

- CWLTool PR: https://github.com/common-workflow-language/cwltool/pull/1233 (Issue: https://github.com/common-workflow-language/cwltool/issues/1232)
- Discourse: https://cwl.discourse.group/t/ask-cwl-to-rename-a-secondary-file/72

### CODE REVIEW

File: tests/secondaryfiles/rename-inputs.cwl [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+          if (secPattern[0] == '^') {
+            var spl = base.split('.');
+            var endIndex = spl.length > 1 ? spl.length - 1 : 1;
+            return resolveSecondary(spl.slice(undefined, endIndex).join("."), secPattern.slice(1));
+          }
+          return base + secPattern;
+        }
+        return [{
+            location: resolveSecondary(self.location, '.accessory'),
+            basename: resolveSecondary(self.basename, '^.accessory')
```
Comment (mr-c) [inline tests/secondaryfiles/rename-inputs.cwl:38]:
```suggestion
        return [{
            "class": "File",
            "location": resolveSecondary(self.location, '.accessory'),
            "basename": resolveSecondary(self.basename, '^.accessory')
```

File: tests/secondaryfiles/rename-outputs.cwl [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+            var spl = base.split(".");
+            var endIndex = spl.length > 1 ? spl.length - 1 : 1;
+            return resolveSecondary(spl.slice(undefined, endIndex).join("."), secPattern.slice(1));
+          }
+          return base + secPattern;
+        }
+        return [
+          {
+            path: resolveSecondary(self.path, "^.accessory"),
+            basename: resolveSecondary(self.basename, ".accessory")
```
Comment (mr-c) [inline tests/secondaryfiles/rename-outputs.cwl:34]:
```suggestion
          {
            "class": "File",
            "path": resolveSecondary(self.path, "^.accessory"),
            "basename": resolveSecondary(self.basename, ".accessory")
```

File: CommandLineTool.yml [hunk 3]
```diff
@@ -55,6 +55,7 @@ $graph:
           [ResourceRequirement](#ResourceRequirement) may now request fractional CPUs.
       * (CommandLineTool)[#CommandLineTool] can now express `intent`
         with an identifier for the type of computational operation.
+      * Added conformance tests to ensure a `secondaryFiles` expression accepts a File object.
```
Comment (illusional) [inline CommandLineTool.yml:58]: @tetron, as requested.

**Comment by tetron** [comment]
I pushed some changes, the input rename test needs to work a little differently to conform to the existing spec, you should probably look at what changed.  Also added text to clarify what values are available to the expression and how different expression return values should be handled.

**Pull request merged**